README.md
# Boxify Theme for Grav

![Boxify](assets/readme_1.png)

Boxify is a stylish HTML5/CSS3 template that was carefully crafted and enhanced with some smooth effects. It’s fitting for any type of portfolio or start-up website, but it’s very flexible and can be used for many other projects. It's made by [Peter Finlan for Codrops](http://tympanus.net/codrops/?p=22554).

# Built on:

* Bootstrap
* Flickity
* fancyBox
* jQuery Waypoints
* Animate.css

# Features:

* HTML5 and CSS3
* Fully Responsive
* Attractive and modern design
* Various templates for presenting your content

# Installation

Installing the Boxify theme can be done in one of two ways. Our GPM (Grav Package Manager) installation method enables you to quickly and easily install the theme with a simple terminal command, while the manual method enables you to do so via a zip file.

The theme by itself is useful, but you may have an easier time getting up and running by installing a skeleton. The [Boxify Site Skeleton](https://github.com/getgrav/grav-skeleton-boxify-site) is a self-contained repository for a complete sites which includes: sample content, configuration, theme, and plugins.

## GPM Installation (Preferred)

The simplest way to install this theme is via the [Grav Package Manager (GPM)](http://learn.getgrav.org/advanced/grav-gpm) through your system's Terminal (also called the command line).  From the root of your Grav install type:

    bin/gpm install boxify

This will install the Boxify theme into your `/user/themes` directory within Grav. Its files can be found under `/your/site/grav/user/themes/boxify`.

## Manual Installation

To install this theme, just download the zip version of this repository and unzip it under `/your/site/grav/user/themes`. Then, rename the folder to `boxify`. You can find these files either on [GitHub](https://github.com/getgrav/grav-theme-boxify) or via [GetGrav.org](http://getgrav.org/downloads/themes).

You should now have all the theme files under

    /your/site/grav/user/themes/boxify

>> NOTE: This theme is a modular component for Grav which requires the [Grav](http://github.com/getgrav/grav), [Error](https://github.com/getgrav/grav-theme-error) and [Problems](https://github.com/getgrav/grav-plugin-problems) plugins.

# Updating

As development for the Boxify theme continues, new versions may become available that add additional features and functionality, improve compatibility with newer Grav releases, and generally provide a better user experience. Updating Boxify is easy, and can be done through Grav's GPM system, as well as manually.

## GPM Update (Preferred)

The simplest way to update this theme is via the [Grav Package Manager (GPM)](http://learn.getgrav.org/advanced/grav-gpm). You can do this with this by navigating to the root directory of your Grav install using your system's Terminal (also called command line) and typing the following:

    bin/gpm update boxify

This command will check your Grav install to see if your Boxify theme is due for an update. If a newer release is found, you will be asked whether or not you wish to update. To continue, type `y` and hit enter. The theme will automatically update and clear Grav's cache.

## Manual Update

Manually updating Boxify is pretty simple. Here is what you will need to do to get this done:

* Delete the `your/site/user/themes/boxify` directory.
* Download the new version of the Boxify theme from either [GitHub](https://github.com/getgrav/grav-theme-boxify) or [GetGrav.org](http://getgrav.org/downloads/themes).
* Unzip the zip file in `your/site/user/themes` and rename the resulting folder to `boxify`.
* Clear the Grav cache. The simplest way to do this is by going to the root Grav directory in terminal and typing `bin/grav clear-cache`.

> Note: Any changes you have made to any of the files listed under this directory will also be removed and replaced by the new set. Any files located elsewhere (for example a YAML settings file placed in `user/config/themes`) will remain intact.

# Setup

If you want to set Boxify as the default theme, you can do so by following these steps:

* Navigate to `/your/site/grav/user/config`.
* Open the **system.yaml** file.
* Change the `theme:` setting to `theme: boxify`.
* Save your changes.
* Clear the Grav cache. The simplest way to do this is by going to the root Grav directory in Terminal and typing `bin/grav clear-cache`.

Once this is done, you should be able to see the new theme on the frontend. Keep in mind any customizations made to the previous theme will not be reflected as all of the theme and templating information is now being pulled from the **boxify** folder.


CHANGELOG.md
# v1.2.1
## 01/15/2021

1. [](#improved)
    * Fixed autoescaping in preparation for Grav 1.7

# v1.2.0
## 03/21/2019

1. [](#new)
    * Set Dependency of Grav 1.5.10+ which has support for new **Deferred Block** Twig extension
    * Implement assets rendering using **Deferred Block** Twig extension 
    
# v1.1.1
## 08/31/2016

1. [](#bugfix)
    * Fixed quotes in jQuery selector throwing JS errors, also in minified js
    * Load external resources using https to prevent errors on SSL enabled sites (demo included)

# v1.1.0
## 07/14/2016

1. [](#improved)
    * Remove unneeded streams from Theme YAML
    * Delete unused composer.json
1. [](#bugfix)
    * Fix setting the page language in the html tag

# v1.0.0
## 09/04/2015

1. [](#new)
    * ChangeLog started...


js/min/scripts-min.js
$(document).ready(function(){$(".wp1").waypoint(function(){$(".wp1").addClass("animated fadeInLeft")},{offset:"75%"}),$(".wp2").waypoint(function(){$(".wp2").addClass("animated fadeInDown")},{offset:"75%"}),$(".wp3").waypoint(function(){$(".wp3").addClass("animated bounceInDown")},{offset:"75%"}),$(".wp4").waypoint(function(){$(".wp4").addClass("animated fadeInDown")},{offset:"75%"}),$("#featuresSlider").flickity({cellAlign:"left",contain:!0,prevNextButtons:!1}),$("#showcaseSlider").flickity({cellAlign:"left",contain:!0,prevNextButtons:!1,imagesLoaded:!0}),$(".youtube-media").on("click",function(a){var b=$(window).width();if(!(b<=768))return $.fancybox({href:this.href,padding:4,type:"iframe",href:this.href.replace(new RegExp("watch\\?v=","i"),"v/")}),!1})}),$(document).ready(function(){$("a.single_image").fancybox({padding:4})}),$(".nav-toggle").click(function(){$(this).toggleClass("active"),$(".overlay-boxify").toggleClass("open")}),$(".overlay ul li a").click(function(){$(".nav-toggle").toggleClass("active"),$(".overlay-boxify").toggleClass("open")}),$(".overlay").click(function(){$(".nav-toggle").toggleClass("active"),$(".overlay-boxify").toggleClass("open")}),$('a[href*="#"]:not([href="#"])').click(function(){if(location.pathname.replace(/^\//,"")===this.pathname.replace(/^\//,"")&&location.hostname===this.hostname){var a=$(this.hash);if(a=a.length?a:$("[name="+this.hash.slice(1)+"]"),a.length)return $("html,body").animate({scrollTop:a.offset().top},2e3),!1}});

js/scripts.js
$(document).ready(function() {

	/***************** Waypoints ******************/

	$('.wp1').waypoint(function() {
		$('.wp1').addClass('animated fadeInLeft');
	}, {
		offset: '75%'
	});
	$('.wp2').waypoint(function() {
		$('.wp2').addClass('animated fadeInDown');
	}, {
		offset: '75%'
	});
	$('.wp3').waypoint(function() {
		$('.wp3').addClass('animated bounceInDown');
	}, {
		offset: '75%'
	});
	$('.wp4').waypoint(function() {
		$('.wp4').addClass('animated fadeInDown');
	}, {
		offset: '75%'
	});

	/***************** Flickity ******************/

	$('#featuresSlider').flickity({
		cellAlign: 'left',
		contain: true,
		prevNextButtons: false
	});

	$('#showcaseSlider').flickity({
		cellAlign: 'left',
		contain: true,
		prevNextButtons: false,
		imagesLoaded: true
	});

	/***************** Fancybox ******************/

	$(".youtube-media").on("click", function(e) {
		var jWindow = $(window).width();
		if (jWindow <= 768) {
			return;
		}
		$.fancybox({
			href: this.href,
			padding: 4,
			type: "iframe",
			'href': this.href.replace(new RegExp("watch\\?v=", "i"), 'v/'),
		});
		return false;
	});

});

$(document).ready(function() {
	$("a.single_image").fancybox({
		padding: 4,
	});
});

/***************** Nav Transformicon ******************/

/* When user clicks the Icon */
$(".nav-toggle").click(function() {
	$(this).toggleClass("active");
	$(".overlay-boxify").toggleClass("open");
});

/* When user clicks a link */
$(".overlay ul li a").click(function() {
	$(".nav-toggle").toggleClass("active");
	$(".overlay-boxify").toggleClass("open");
});

/* When user clicks outside */
$(".overlay").click(function() {
	$(".nav-toggle").toggleClass("active");
	$(".overlay-boxify").toggleClass("open");
});

/***************** Smooth Scrolling ******************/

$('a[href*="#"]:not([href="#"])').click(function() {
	if (location.pathname.replace(/^\//, '') === this.pathname.replace(/^\//, '') && location.hostname === this.hostname) {

		var target = $(this.hash);
		target = target.length ? target : $('[name=' + this.hash.slice(1) + ']');
		if (target.length) {
			$('html,body').animate({
				scrollTop: target.offset().top
			}, 2000);
			return false;
		}
	}
});
